main/src/ca/uwaterloo/flix/language/ast/AtomicOp.scala

```diff
@@ -62,9 +62,9 @@ object AtomicOp {
 
   case class StructNew(sym: Symbol.StructSym, fields: List[Name.Label]) extends AtomicOp
 
-  case class StructPut(sym: Symbol.StructSym, field: Name.Label) extends AtomicOp
+  case class StructPut(sym: Symbol.StructSym, idx: Int, field: Name.Label) extends AtomicOp
```

Comment:
Remind me, why do we not have a FieldSym? I would think the index could be part of the sym.

Reply:
Jonathan agrees.

Reply:
Yea, I am currently working on a PR that adds back the fieldsym. I think, though, that would be out of the scope of this one